### PR TITLE
Allow the build scan plugin to read environment variables and reenable test

### PR DESCRIPTION
**Pull request by mlopatkin**
Some versions of the build scan plugin read environment variables during the configuration phase but do not take decisions upon them, so recording these variables as inputs to the configuration cache is a false positive.
Similar to system properties, we skip these accesses. The escape hatch to disable the workarounds is provided to make fixing the issues in the scan plugin easier in the future. Workarounds for environment variables and system properties can be turned off separately.

### CODE REVIEW

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt [hunk 1]
```diff
@@ -31,7 +31,21 @@ object Workarounds {
...
+    }
+
+    private
+    fun isBuildScanPlugin(from: String): Boolean {
         return from.startsWith("com.gradle.scan.plugin.internal.")
     }
+
+    private
+    fun shouldDisableInputWorkarounds(): Boolean {
+        return System.getProperty("org.gradle.internal.disable.input.workarounds")?.equals("true", ignoreCase = true) ?: false
```
Comment (eskatos) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt:49]: :question: Won't this trip over the instrumentation and be reported as an input?

Comment (mlopatkin) [reply subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt:49]: I don't think we instrument classes in `configuration-cache`, at least that is what my experiments with debugger show.

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Review by eskatos — APPROVED** [review]
:+1:

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**